### PR TITLE
Port to netstandard

### PR DESCRIPTION
**Pull request by nbarbettini**
Progress on porting the SDK to .NET Standard.

(Best to look at the diff with [whitespace ignored](https://github.com/okta/oktasdk-csharp/pull/30/files?w=1)).

The new `Okta.Core.Client.csproj` project targets .NET Standard 1.1, which means it is usable from .NET Framework 4.5 *and* .NET Core projects. This does mean we will need to drop support for Silverlight 5 and .NET Framework 4.0. Since this is potentially a breaking change, I've incremented the major version number to 1.0.0.

Outstanding todos:

* All of the tests that rely on SQL Express will fail currently. We'll need to figure out some other way of doing data-driven tests.
* I think the Automation (PowerShell cmdlet) project should be moved to its own solution and repo. If it needs to rely on the SDK, it should do so through a public NuGet package reference.

### CODE REVIEW

File: src/Okta.Core.Client/RegexUtils.cs [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Globalization;
+
+namespace Okta.Core
+{
+    /// <summary>
+    /// Regular Expressions utilities
+    /// </summary>
+    public class RegexUtilities
```
Comment (jmelberg-okta) [inline src/Okta.Core.Client/RegexUtils.cs]: Thinking we should remove this class. It is only used for unit tests and doesn't need to be in the core project.

File: src/Okta.Core.Client/OktaJsonConverter.cs [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
+
+                                // Add it to our dictionary
+                                linkDictionary.Add(linkJTokenName, linkList);
+
+                            }
+
+                            v = linkDictionary;
+                        }
+
+                        //2016-01-29, RL, bug fix to handle the Hints dictionary case
```
Comment (jmelberg-okta) [inline src/Okta.Core.Client/OktaJsonConverter.cs]: We have massive commented code. We should clean this up

File: src/Okta.Core.Client/Models/Users/Provider.cs [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    /// </summary>
+    public class Provider : ApiObject
+    {
+        [JsonProperty("type")]
+        public string Type { get; set; }
+
+        [JsonProperty("name")]
+        public string Name { get; set; }
+    }
+}
```
Comment (jmelberg-okta) [inline src/Okta.Core.Client/Models/Users/Provider.cs]: nit: Newline

File: src/Okta.Core.Client/Models/MFA/FactorProviderType.cs [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    /// </summary>
+    public static class FactorProviderType
+    {
+        public const string Okta = "OKTA";
+        public const string Symantec = "SYMANTEC";
+        public const string Google = "GOOGLE";
+        public const string Duo = "DUO";
+        public const string RSA = "RSA";
+        public const string Yubico = "YUBICO";
+
```
Comment (jmelberg-okta) [inline src/Okta.Core.Client/Models/MFA/FactorProviderType.cs]: nit: Extra line

File: src/Okta.Core.Client/Models/Apps/App.cs [hunk 5]
```diff
@@ -0,0 +1,305 @@
...
+                OptionalField2 = optionalField2, 
+                OptionalField2Value = optionalField2Value, 
+                OptionalField3 = optionalField3, 
+                OptionalField3Value = optionalField3Value
+            };
+
+            return app;
+        }
+
+        /*// Build SAML 2.0
```
Comment (jmelberg-okta) [inline src/Okta.Core.Client/Models/Apps/App.cs]: @lboyette-okta: Any idea why we have this comment block?

Comment (lboyette-okta) [inline src/Okta.Core.Client/Models/Apps/App.cs]: I don't think I got it working. Removing it's the right thing to do

**Comment by nbarbettini** [comment]
I'm not going to spend any time on this, since it would have been a breaking change and is superseded by 1.x. 0.x won't be compatible with netstandard unless we decide to revisit this.

**Pull request closed without merging**